lib/assets/javascripts/views/answers/rda_metadata.js

```diff
@@ -0,0 +1,406 @@
+  // initializes the standards property to the result of an AJAX POST
+  function getStandards(name, num, child) {
+    // slice -4 from url to remove '/api/'
+    noWaiting += 1;
+    $.ajax({
+      url: `${url.slice(0, -4)}query/schemes`,
+      type: 'POST',
+      crossDomain: true,
+      data: `keyword=${name}`,
+      dataType: 'json',
+    }).done((result) => {
+      if (isUndefined(child)) {
+        minTree[num].standards = result.ids;
+      } else {
+        minTree[num].children[child].standards = result.ids;
+      }
+      if (result.ids.length < 1) {
+        removeUnused(name);
+      }
+      noWaiting -= 1;
+      initDescriptions(result.ids);
+    });
+  }
+
+  // clean up the data initially returned from the API
+  function cleanTree(apiTree) {
+    // iterate over api_tree
+    Object.keys(apiTree).forEach((num) => {
+      minTree[num] = {};
+      minTree[num].name = apiTree[num].name;
+      minTree[num].children = [];
+      if (apiTree[num].children !== undefined) {
+        Object.keys(apiTree[num].children).forEach((child) => {
+          minTree[num].children[child] = {};
+          minTree[num].children[child].name = apiTree[num].children[child].name;
+          minTree[num].children[child].standards = [];
+          getStandards(minTree[num].children[child].name, num, child);
+        });
+      }
+      // init a standards on top level
+      minTree[num].standards = [];
+      getStandards(minTree[num].name, num, undefined);
+    });
+  }
+
+
+  // create object for typeahead
+  function initTypeahead() {
+    const data = [];
+    const simpdat = [];
+    Object.keys(descriptions).forEach((id) => {
+      data.push({ value: descriptions[id].title, id });
+      simpdat.push(descriptions[id].title);
+    });
+    const typ = $('.standards-typeahead');
+    typ.typeahead({ source: simpdat });
+  }
+
+  function initStandards() {
+    // for each metadata question, init selected standards according to html
+    $('.rda_metadata').each(function () {
+      // list of selected standards
+      const selectedStandards = $(this).find('.selected_standards .list');
+      // form listing of standards
+      const formStandards = $(this).next('form').find('#standards');
+      // need to pull in the value from frm_stds
+      const standardsArray = JSON.parse(formStandards.val());
+      // init the data value
+      formStandards.data('standard', standardsArray);
+      Object.keys(standardsArray).forEach((key) => {
+        // add the standard to list
+        if (key === standardsArray[key]) {
+          selectedStandards.append(`<li class="${key}">${key}<button class="remove-standard"><i class="fa fa-times-circle"></i></button></li`);
+        } else {
+          selectedStandards.append(`<li class="${key}">${descriptions[key].title}<button class="remove-standard"><i class="fa fa-times-circle"></i></button></li>`);
+        }
+      });
+    });
+  }
+
+  function waitAndUpdate() {
```

Comment:
What happens when the API request to the RDA API fails or takes an exceptionally long time? Does the user know something is happening in the background? The `noWaiting` variable doesn't get set back to zero until you receive a successful response form the API call.

You could also have probably used @jollopre `utils/debounce.js` for this which provides a timer like this along with a way to cancel it.